### DISCUSSION TITLE
WYSIWYG csv table editor

### DISCUSSION THREAD

**Discussion by GavinPen** [GitJournal/GitJournal#681] [Feature Requests]
I know CSVs aren't markdown, but a WYSIWYG csv table editor would be a great addition to GitJournal.
I don't know of any mobile app that does this.
This would help allow a csv file to be used as a todo list (you could easily edit a 'done' column)
I find CSV syntax more intuitive than markdown table syntax, but still unwieldy on a phone.
GitHub already allows simple filtering of csv files via the 'search this file' box (I've filtered by 'disaster' in the screenshot)

![GitHub - search this file](https://user-images.githubusercontent.com/9166529/159350276-1f516bfa-efc5-4eb0-a067-bcf181ff0fc3.png)

**Comment by vHanda** [comment]
Damn, this sounds like a fun challenge. Both figuring out what kind of UI would work best on a mobile and implementing it.

However, my inner product manager would be quite upset if I prioritized this over the bugs, and other major features, so this will probably have to wait.

Though, if this got some 10-15 👍🏽s, then it would put it very near the top of my list.

**Comment by GavinPen** [comment]
Thanks @vHanda - I do understand why this wouldn't make the high priority list, unless we get lots of upvotes!
There are Android csv editors out there, but I don't know of any which automatically work with git.

(It's possible, though not very pretty, for a csv file to also work as a markdown checklist

eg:
- [ ] ,a
- [ ] ,b
- [ ] ,c

However, if I save this as a .csv, GitJournal won't open it.
And if I save it as .md, GitHub won't provide the 'search this file' filtering!)

**Comment by vHanda** [comment]
If you go to Settings -> storage file formats -> note file types, you can configure GitJournal to open CSV files. It will open then in the Raw editor, but it should work.

**Comment by GavinPen** [comment]
thanks. I have now been able to open a csv this way.
However, I can't see the .xlsx file in my folder, despite checking the .xlsx box under 'note file types'.
Is the csv / xlsx compatibility mentioned anywhere else in the GitJournal docs? (I couldn't find it)

**Comment by GavinPen** [comment]
A todo.csv could be a more portable alternative to todo.txt.
A csv's columns, more so than todo.txt's tag-based fields, are easily parsed by lots of applications.
A 'done' column in a todo.csv could be displayed as a checkbox in GitJournal
